src/runtime/threads/policies/parse_affinity_options.cpp

```diff
@@ -842,6 +843,72 @@ namespace hpx { namespace threads { namespace detail
     }
 
     ///////////////////////////////////////////////////////////////////////////
+    void decode_numabalanced_distribution(hwloc_topology_info& t,
+        std::vector<mask_type>& affinities,
+        std::size_t used_cores, std::size_t max_cores,
+        std::vector<std::size_t>& num_pus, error_code& ec)
+    {
+        std::size_t num_threads = affinities.size();
+        num_pus.resize(num_threads);
+        // numa nodes
+        std::size_t num_numas = (std::max)(std::size_t(1),t.get_number_of_numa_nodes());
```

Comment:
Should we add a fallback to the number of sockets here if the kernel misses NUMA support? We do that already in a couple of places.

Reply:
Not really any point. if the kernel lacks numa support then asking for numa-balanced is meaningless anyway. Let it default to 1 domain.

Reply:
What about reporting an error at startup if the user asks for NUMA-domain awareness on a machine which does not have NUMA domains.